### DISCUSSION TITLE
What will make it a success?

### DISCUSSION THREAD

**Discussion by craigyjp** [probonopd/MiniDexed#167] [Ideas]
I love this project, although I cannot dedicate anytime to developing it myself I have two ideas that will make it into a killer project.

1. Lack of sysex dumps for sending and receiving patches over MIDI and being able to store those patches into any bank/locations
2. Lack of editing over MIDI, it's a great device but we really need to edit it over MIDI from a standard DX7 format editor and agin to be able to save those edits to any bank/location.

**Comment by dcoredump** [comment]
I hope I can find time the next days for exactly those two points. The code exists in MicroDexed. I have to move it into Synth_Dexed and than MiniDexed can use this feature also. Please give me a few more days. I unfortunately can not make it so as fast as I want, because I have to do some paid work, so that beer is in the fridge :)

**Reply by craigyjp** [reply]
No pressure from me, take your time.

**Reply by probonopd** [reply]
This will make a lot of people very happy @dcoredump :+1: take your time.

**Reply by dcoredump** [reply]
> This will make a lot of people very happy @dcoredump :+1: take your time.

I started my work on this. It's a little bit difficult to make the code working for both platforms. But I am sure at least it will work 😀

**Comment by now-its-dark** [comment]
One note regarding software editors, just wanted to call this out—

There are some great TX816-specific editors/librarians, which can manage all the parts and assign patches to each of them from a single interface. It would be ideal to eventually support these, since that capability is not available using a normal DX7 editor (without some workarounds).  

[SynLib DX-TX for Windows / Atari ST](http://m.bareille.free.fr/synlibdxtx/synlibdxtx.htm) (this one is free and will also work under WINE)

[Patchbase TX816 for iOS / Mac](https://coffeeshopped.com/patch-base/editor/yamaha/tx816)

Thanks so much to everyone involved in making MiniDexed a reality, it is such an exciting project! 😃 🙏

**Reply by probonopd** [reply]
Not forge the open sourced [Dexed](https://asb2m10.github.io/dexed/), which runs standalone or as a DAW plugin on Windows, Mac, and Linux. Once @dcoredump finds some time to teach MiniDexed how to [receive Sysex](https://github.com/probonopd/MiniDexed/issues/82), it will be possible to use such software.

**Comment by dcoredump** [comment]
Hm, just thinking about a problem for me to test my SYSEX support: I have no equipment which can produce SYSEX data as a USB host, because the RPi had only non-host USB ports.
Only adding a dedicated DIN-MIDI would help to connect a computer to MiniDexed.

How do you connect e.g. an editor- software to MiniDexed?

**Reply by miotislucifugis** [reply]
I believe that you would need to use a usb to din midi converter  and connect the OUT from that to minidexed din serial Midi IN.
A lot of the cheap/small  usb to midi converters cannot handle sysex tho' (at least with preset dumps)... I think that the buffers are too small.    Previously, Ive had to use my MOTU audio interface which has midi In and Out to send sysex to vintage DIN-only gear.  Non of the dongle-type adapters  would work.

**Reply by probonopd** [reply]
@dcoredump 

> How do you connect e.g. an editor- software to MiniDexed?

I am using a Raspberry Pi Pico that runs firmware that turns it into a MIDI device. It outputs serial data which I feed to the TX and RX lines.

https://github.com/probonopd/MiniDexed/wiki/Hardware#usb-midi-device

if you have a build that I should test, I'm happy to do it, in case you don't have a Raspberry Pi Pico available.
